tests/unittest/_torch/thop/test_w4a8_linear.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+import torch
+from torch.nn.parameter import Parameter
+
+import tensorrt_llm.quantization.functional
+from tensorrt_llm._torch.autotuner import autotune
+from tensorrt_llm._torch.custom_ops.torch_custom_ops import \
+    finegrainedMixedDtypeGemm
+from tensorrt_llm._torch.modules.linear import Linear
+from tensorrt_llm._utils import get_sm_version
+from tensorrt_llm.models.modeling_utils import QuantAlgo, QuantConfig
+
+
+@pytest.mark.parametrize("weights_dtype", [torch.uint8])
+@pytest.mark.parametrize(
+    "dtype",
+    [torch.float16],
+)
+def test_w4a8_linear(dtype, weights_dtype, has_zero=False):
+
+    if get_sm_version() > finegrainedMixedDtypeGemm.MAX_SUPPORTED_SM_VERSION:
+        pytest.skip(
+            f"W4A16/W4A8 is not supported in this SM version {get_sm_version()}"
+        )
+
+    SEQ_LEN = 10
+    HIDDEN_SIZE = 128
+    GROUP_SIZE = 128
+    torch.manual_seed(0)
+
+    total_groups = (HIDDEN_SIZE + GROUP_SIZE - 1) // GROUP_SIZE
+
+    x = torch.randn((SEQ_LEN, HIDDEN_SIZE), dtype=dtype).cuda()
+    w = torch.randint(0,
+                      2**32 - 1, (HIDDEN_SIZE, HIDDEN_SIZE // 8),
```

Comment:
I ran this test with weight shape `input_dim != output_dim` and it appears to fail. Could you add a case with, say, output_dim=384 and see if it passes?